decidim-census/spec/helpers.rb

```diff
@@ -0,0 +1,11 @@
+module Decidim
+  module Census
+    module Helpers
+      def encode_id_document(id_document)
```

Comment:
I see this same code repeated several times in the codebase:
- Two times at migrations
- Inside the `normalize_and_encode...` method
- Inside the `uniqe_id` method

 Is there any reason to not re-use it in different places?

Reply:
These are the reasons for every occurrence:

- **Migrations**: Adding dependencies to a migration is dangerous. We can make the migrations fail in the future without noticing it as migrations are not being tested. And when installing the engine the failing migrations will be copied to the main project.
- **normalize_and_encode**: It is where the encoding is actually happending.
- **unique_id (on the decidim-census)**:  Is already using the normalize_and_encode results.
- **unique_id (on the decidim-diba_census_api)**: I don't want to make the *decidim-diba_census_api* dependant on the *decidim-census* library. The alternative is extracting the  encoding code to a separate gem and make it required by both engines. However, I decided that repeating the code was preferred over adding complexity to the whole system.
- **specs**: I didn't want to make a test helper method rely on code being used by the project. If I do so, then the specs won't be as reliable.

Reply:
I imagined that it was not an oversight. Go ahead!